es-core/src/Window.cpp

```diff
@@ -269,16 +269,13 @@ void Window::render()
 	{
 		mInfoPopup->render(transform);
 	}
-	
+
 	if(mTimeSinceLastInput >= screensaverTime && screensaverTime != 0)
 	{
-		if (!isProcessing() && mAllowSleep && (!mScreenSaver || mScreenSaver->allowSleep()))
-		{
-			// go to sleep
-			if (mSleeping == false) {
-				mSleeping = true;
-				onSleep();
-			}
+		unsigned int systemSleepTime = (unsigned int)Settings::getInstance()->getInt("SystemSleepTime");
+		if(!isProcessing() && mAllowSleep && systemSleepTime != 0 && mTimeSinceLastInput >= systemSleepTime) {
```

Comment:
Apologies if it's somewhere else, but it's not clear to me why are removing "(!mScreenSaver || mScreenSaver->allowSleep())" ?
If I read it correctly, those are there for ES to only go to sleep if either the screensaver isn't active, or if the screensaver allows sleeping. Won't this change the behavior and always force sleep over the screensaver?
Maybe that is intended, but wanted to probe into this.

Reply:
D'oh. Somehow my comment for resolving did not make it through. Retyping it again.

The initial confusion may be caused by the fact that I pm'ed jdrassa about the sleep/wakeup events for the wiki page (Aug. 2019). Thus it looked like it was completly implemented, but it wasn't.

For your challenge in this code segment: Good to have strict reviews. It was intented to enter sleep without having an screensaver set.
If you have a screensaver enabled it is not possible via UI to set sleep timeout < screensaver timeout. The user will get a note and in turn the timeout is set that sleep and screensaver have the same value. However, it is possible to force sleep timeout < screensaver timeout via the command line, but this is a different story: I assume someone knows what he/she is doing when editing files manually.

You can set sleep timeout to 0 which has the same semantic as in the screensaver (=off).

Reply:
Thanks!